core/amqp_leader-1.0.0/src/amqp_leader_listener.erl

```diff
@@ -0,0 +1,241 @@
+    {'noreply', State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling all non call/cast messages
+%%
+%% @spec handle_info(Info, State) -> {noreply, State} |
+%%                                   {noreply, State, Timeout} |
+%%                                   {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_info(any(), state()) -> handle_info_ret_state(state()).
+handle_info(_Info, State) ->
+    lager:warning("~s unhandled info ~p", [node(), _Info]),
+    {'noreply', State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Allows listener to pass options to handlers
+%%
+%% @spec handle_event(JObj, State) -> {reply, Options}
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_event(wh_json:object(), state()) -> {'reply', []}.
+handle_event(JObj, #state{name = Name}) ->
+    wh_util:put_callid(wapi_leader:queue(Name, node())),
+    NodeBin = iolist_to_binary(io_lib:format("~s", [node()])),
+    case wh_json:get_value(<<"Node">>, JObj) of
+        NodeBin -> 'ok';
+        _ ->
+            Msg = erlang:binary_to_term(wh_util:from_hex_binary(wh_json:get_value(<<"Message">>, JObj))),
+            Name ! Msg
+    end,
+    {'reply', []}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% This function is called by a gen_server when it is about to
+%% terminate. It should be the opposite of Module:init/1 and do any
+%% necessary cleaning up. When it returns, the gen_server terminates
+%% with Reason. The return value is ignored.
+%%
+%% @spec terminate(Reason, State) -> void()
+%% @end
+%%--------------------------------------------------------------------
+-spec terminate(any(), state()) -> 'ok'.
+terminate(_Reason, _State) ->
+    lager:debug("listener terminating: ~p", [_Reason]).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Convert process state when code is changed
+%%
+%% @spec code_change(OldVsn, State, Extra) -> {ok, NewState}
+%% @end
+%%--------------------------------------------------------------------
+-spec code_change(any(), state(), any()) -> {'ok', state()}.
+code_change(_OldVsn, State, _Extra) ->
+    {'ok', State}.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+-spec recv_ready(reference()) -> 'true'.
+recv_ready(Ref) ->
+    receive
+        {Ref, 'ready'} -> ready_when_node_is_up()
+    after
+        1 * ?MILLISECONDS_IN_SECOND ->
+            lager:debug("listener not ready yet"),
+            recv_ready(Ref)
+    end.
+
+-spec ready_when_node_is_up() -> 'true'.
+ready_when_node_is_up() ->
+    case ets:match('wh_nodes', #wh_node{node = node(), expires = '$2', _ = '_'}) of
```

Comment:
@siplabs looks like its still using wh_nodes ets table, and I didn't see the update to wh_nodes module.  Perhaps there is a push to this branch that is missing?